validator-auction-rest-server/lib/EthEventsClient.js

```diff
@@ -0,0 +1,169 @@
+        ])
+
+        result.whitelistedAddresses = whitelistedAddresses
+        result.bids = bids
+        result.bids.forEach(bid => {
+            bid.currentPrice = EthEventsClient.getCurrentPrice(auctionStart * 1000, bid.timestamp * 1000)
+        })
+        result.takenSlotsCount = result.bids.length
+        result.freeSlotsCount = MAX_SLOTS_COUNT - result.takenSlotsCount
+        result.remainingSeconds = EthEventsClient.calculateRemainingAuctionSeconds(auctionStart, currentBlockTime)
+        result.currentPrice = EthEventsClient.getCurrentPrice(auctionStart * 1000, currentBlockTime * 1000)
+        return result
+    }
+
+    async getCurrentBlockTime() {
+        const response = await axios.post(`${this._baseUrl}/block/search/`, {
+            'query': {
+                'match_all': {}
+            },
+            'size': 1,
+            'sort': [
+                {
+                    'number.num': {
+                        'order': 'desc'
+                    }
+                }
+            ],
+            '_source': 'timestamp'
+        }, this._axisConfig)
+        if(response.data.hits.total > 0) {
+            return response.data.hits.hits[0]._source.timestamp
+        }
+        else {
+            throw new Error('Could not get block time')
+        }
+    }
+
+    async getAuctionStartInSeconds() {
+        const response = await axios.post(`${this._baseUrl}/event/search/`, {
+            'size': 1,
+            'query': {
+                'bool': {
+                    'must': [
+                        {
+                            'term': {
+                                'address.raw': this._contractAddress
+                            }
+                        },
+                        {
+                            'term': {
+                                'event.raw': 'AuctionStarted'
+                            }
+                        }
+                    ]
+                }
+            },
+            '_source': 'args'
+        }, this._axisConfig)
+        return response.data.hits.total > 0 ? response.data.hits.hits[0]._source.args[0]['value.num'] : 0
+    }
+
+    async getBidEvents() {
+        const response = await axios.post(`${this._baseUrl}/event/search/`, {
+            'size': 1000,
+            'query': {
+                'bool': {
+                    'must': [
+                        {
+                            'term': {
+                                'address.raw': this._contractAddress
+                            }
+                        },
+                        {
+                            'term': {
+                                'event.raw': 'BidSubmitted'
+                            }
+                        }
+                    ]
+                }
+            },
+            '_source': 'args'
+        }, this._axisConfig)
+
+        return response.data.hits.hits.map(hit => {
+            // better check if they are in that sequence
```

Comment:
Can we not better query them by name? This is safer I think. 
The event already changed to currently
`BidSubmitted(address bidder, uint bidValue, uint slotPrice, uint timestamp)`